homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,257 @@
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.wifi_uptime if data.networking else None,
+        translation_key="wifi_uptime",
+        key="wifi_uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.mqtt_uptime if data.networking else None,
+        translation_key="mqtt_uptime",
+        key="mqtt_uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.telemetry.water_temperature
+        if data.telemetry
+        else None,
+        translation_key="water_temperature",
+        key="water_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.telemetry.pressure if data.telemetry else None,
+        translation_key="water_pressure",
+        key="water_pressure",
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: (data.telemetry.flow / 1000)
+        if (data.telemetry and data.telemetry.flow is not None)
+        else None,
+        translation_key="water_flow_rate",
+        key="water_flow_rate",
+        native_unit_of_measurement=UnitOfVolumeFlowRate.LITERS_PER_MINUTE,
+        device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.uptime if data.state else None,
+        translation_key="uptime",
+        key="uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.power_supply if data.state else None,
+        translation_key="power_supply",
+        key="power_supply",
```

Comment:
what does this sensor actually deliver as value?

Reply:
If the device is running on batteries on mains - it is very important for users to display that information.

Reply:
If it's two values => it should be migrated to a binary_sensor.
If it's more than two values => could it be an enum?

Reply:
No it is not binary sensor. According to API it has three states: Mains powered, Battery, Mains powered + battery. I can convert it to the enum, but I have a lot of more changes in library so I would like to 'enum it' in next PR.

Reply:
We can convert it to two separate binary sensors like
Batteries -> yes or no
Mains power -> yes or no